### DISCUSSION TITLE
xDrip causes persistent sensor error message on Dexcom G6 (Android 12).

### DISCUSSION THREAD

**Discussion by kbehel** [NightscoutFoundation/xDrip#1903] [Q&A]
xDrip causes persistent sensor error message on Dexcom G6 (Android 12). Disabling bluetooth or rebooting phone fixes it. 

Is this a known issue? 
I just enabled "Close GATT on BLE disconnect" to see if it fixes things, though I don't even know if the Bluetooth watchdog is the cause of the sensor errors

I'll get the sensor error message from xDrip and on my Dexcom receiver (using both at the moment). Error doesn't go away after an hour.

Pixel 5a,  Android 12

<img width="350" src="https://user-images.githubusercontent.com/22627342/148106924-f4addcec-8f44-4cd2-bcf2-af082ce09610.png" /><img width="350" src="https://user-images.githubusercontent.com/22627342/148106928-cb75331a-4fa0-4b60-9ef2-652f15c5c9ff.png" /><img width="350" src="https://user-images.githubusercontent.com/22627342/148106936-d775e0fb-c209-426e-9b9a-c5c6517b681b.png" /><img width="350" src="https://user-images.githubusercontent.com/22627342/148139863-46323826-fcee-4a76-a3ce-849e6d93ae1a.png" />

**Comment by Navid200** [comment]
Would you please have a look at the recommended settings?
Are any of your settings different?
https://github.com/Navid200/xDrip/wiki/G6-Recommended-Settings

**Reply by kbehel** [reply]
those match my settings

**Comment by jamorham** [comment]
`Sensor Errors` is a state message communicated from the Transmitter to xDrip.  This is usually to do with sensor noise. Typically it resolves itself after a period of time (eg an hour)

My guess is that any action taken that appears to resolve the problem is just a coincidence. I don't believe there is anything xDrip can do to cause this error or resolve it.

The only other possibility would be if xDrip were not correctly updating its copy of this state message, but as OP says the same problem occurs on another receiver which makes me think that isn't what is happening and this is just a regular `Sensor Errors` state caused by sensor noise.

To gather more information, I would say next time it appears, reboot the phone as quickly as possible because I would expect that that would not resolve it and would show that it isn't a software bug within the application or phone bluetooth stack. If doing this then keep notes of when the problem appears, when the phone is rebooted and when the problem resolves and any pattern. 

I also would advise blacking out transmitter id's from screenshots for privacy and security reasons.

**Reply by kbehel** [reply]
I've had this error several times now for 30-60 minutes, and the restart (or toggling bluetooth) fixed it within 2-3 minutes. 

and just as I was typing this, it happened again. A reboot and 3 minutes later all is good. 

So there is definitely a strong correlation to reboot or Bluetooth power cycles.  My son has a bit of BLE on Android experience and was suggesting some ways this could go wrong. He had some ideas to try to get some Android logs.

**Reply by jamorham** [reply]
If you continue to believe there is a correlation with resetting bluetooth or rebooting the phone then please try to record more data by enabling the OB1 debug logs: https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector
Once you get to a stage where you have a recently case of sensor errors which were resolved by a reboot etc then use the log upload feature and be sure to include your github name or other identifier and I can review the logs to see if anything stands out.

**Comment by tolot27** [comment]
@kbehel Please add one screenshot of the log showing the problem to your initial post.

**Reply by kbehel** [reply]
Done

**Comment by kbehel** [comment]
More data -  when the next sensor error occurred,  I turned off the Dexcom receiver and the error was still there ~20 mins later.  Turned the Dexcom receiver back on and the error was still there another 20 mins later.
This was with xDrip still active.

**Reply by tolot27** [reply]
And what happens if you disable xDrip completely (in the app settings of Android)?
Does the Dexcom receiver then shows no sensor errors over a longer period of time?

**Reply by kbehel** [reply]
I can try that the next time I get this error.

**Comment by tolot27** [comment]
Please also add more details to your environment (phone model, detailed Android version). Does this problem happend just with this sensor or with previous sensors/xDrip versions, too?

Looks like your sensor expires soon. It would be interesting if the problem reoccurs with a new sensor.

**Comment by kbehel** [comment]
Update - 

Have not had the problem re-occur yet with a different sensor. 

Because of the way tight correlation between error clearing when xDrip was disconnected, I can't just solely blame the sensor. Perhaps there are some error situations that the Dexcom receiver handles better than xDrip. So the sensor could be partially at fault but Dexcom's receiver does a better job at clearing it. Someone I know with experience with BLE and Android had a couple of theories about how that could happen with BLE.

If it starts happening again, I'll try to get more logs.

**Comment by parapenT1sta** [comment]
I had the same issue with the G5 sensor I start using few days ago, stopped  working with xDrip with sensor errors, I switched to G5 app and receiver and no failures so far.
I don't know if there is an option to ignore the errors in xDrip or to no stop at the very first error but I think this could be handy, used with caution for sure.
